examples/medplum-chart-demo/src/bots/core/charting-utils.ts

```diff
@@ -0,0 +1,281 @@
+      encounter: { reference: getReferenceString(encounter) },
+      performer: [{ reference: getReferenceString(user) }],
+      effectiveDateTime: observationData.date,
+    };
+
+    observations.push(observation);
+  }
+
+  return observations;
+}
+
+/**
+ * This function handles adding blood pressure measurements to an observation since there are often both systolic and diastolic
+ * measurements which need to be added to the component element.
+ *
+ * @param observationData - The data object containing the blood pressure values
+ * @returns An Observation component element with diastolic, systolic, or both blood pressure measurements
+ */
+export function handleBloodPressure(observationData: ObservationData): ObservationComponent[] {
+  const components: ObservationComponent[] = [];
+  const bloodPressure = observationData.bloodPressure;
+
+  // If a diastolic measurement exists, add it
+  if (bloodPressure.diastolic) {
+    components.push({
+      code: { coding: [{ code: '8462-4', system: 'http://loinc.org', display: 'Diastolic blood pressure' }] },
+      valueQuantity: {
+        value: bloodPressure.diastolic,
+        unit: 'mm[Hg]',
+      },
+    });
+  }
+
+  // If a systolic measurement exists, add it
+  if (bloodPressure.systolic) {
+    components.push({
+      code: { coding: [{ code: '8480-6', system: 'http://loinc.org', display: 'Systolic blood pressure' }] },
+      valueQuantity: {
+        value: bloodPressure.systolic,
+        unit: 'mm[Hg]',
+      },
+    });
+  }
+
+  return components;
+}
+
+/**
+ * This function takes condition data and turns it into an array of bundle entries that can be used in a batch transaction. If the
+ * Condition is being added to the problem list, then an additional Condition resource will be created. For more details see the
+ * Representing Diagnoses docs here: https://www.medplum.com/docs/charting/representing-diagnoses
+ *
+ * @param conditionData - Data object containg codes and values for a condition/reason for the visit.
+ * @param encounter - The encounter the data is derived from.
+ * @param user - The user creating the Condition resource
+ * @returns An array of bundle entries containing the Condition resource that can be created in a batch request.
+ */
+export function createConditions(conditionData: ConditionData, encounter: Encounter, user: Practitioner): Condition[] {
+  const conditions: Condition[] = [];
+  // const request: BundleEntryRequest = {
+  //   method: 'PUT',
+  //   url: `Condition?encounter=${getReferenceString(encounter)}&code=${code}`,
+  // };
+  // Create a condition for the encounter diagnosis
+  const encounterDiagnosis: Condition = {
+    resourceType: 'Condition',
+    subject: encounter.subject as Reference<Patient>,
+    encounter: { reference: getReferenceString(encounter) },
+    recorder: { reference: getReferenceString(user) },
+    asserter: { reference: getReferenceString(user) },
+    code: conditionData.reasonForVisit,
```

Comment:
can `conditionData` just be of type `Partial<Condition>`, rather than a custom datatype?